### DISCUSSION TITLE
Work piece height Z probing

### DISCUSSION THREAD

**Discussion by Meliusja** [terjeio/ioSender#416] [General]
Just curious, is there a reason work piece height needs to be negative?   My Z probe is mounted in a table slot and I set the probe height to the distance above the table, which is positive.   Then I set workpiece height to the distance from the table to the 0Z position of my stock.  Maybe I'm using this feature wrong but it does work if height is negative.

Jeff Melius

**Comment by terjeio** [comment]
> Just curious, is there a reason work piece height needs to be negative?

I do not understand what you mean, you have to enter a negative value in which field in the UI?

**Reply by Meliusja** [reply]
Sorry for the ugly screen photo, I was in a hurry.  The Work Piece Height shown in blue must be negative.  In the other photo you can see my zprobe attached to the back right corner of the table.  I have this probe height above table entered in the probe height box on the left side of the screen as a positive number.  I then enter the work piece height above the table as a negative number.
Jeff


![image](https://github.com/user-attachments/assets/35cd8360-ab14-4845-a4f5-80c2b0563580)
![image](https://github.com/user-attachments/assets/975a5601-4354-4393-a6e3-7e4ffb6afc7c)

**Comment by Meliusja** [comment]
Maybe the intent was probe on top the work piece and G54 zero set to the table or vice floor.  That would make them both positive numbers.
Jeff

**Reply by terjeio** [reply]
> Maybe the intent was probe on top the work piece and G54 zero set to the table or vice floor. 

Correct. If you want 0 on the top of the work piece set the workpiece height to 0.

**Comment by Meliusja** [comment]
That makes sense thanks for the clarification.  I am using my probe mounted
to the table at my G28 position to probe each time I change tools so that
is why I enter a negative number for work height zero above table.  Maybe
there is a smarter way to do this, but entering a negative work height
seems to work fine.
Jeff
